### PR TITLE
SNC: Ziatora's Envoy and PlayEffect Support

### PR DESCRIPTION
**Pull request by Northmoc**
mild refactoring and tweaking to PlayEffect:
 - "ForgetTargetRemembered" -> "ForgetPlayed" should be easier for scripters to parse
 - made "ForgetPlayed" work for Lands as well

### CODE REVIEW

**Review by Agetian — APPROVED** [review]

File: forge-gui/res/cardsfolder/upcoming/ziatoras_envoy.txt [hunk 1]
```diff
@@ -0,0 +1,15 @@
+Name:Ziatora's Envoy
+ManaCost:1 B R G
+Types:Creature Viashino Warrior
+PT:5/4
+K:Trample
+T:Mode$ DamageDone | ValidSource$ Card.Self | ValidTarget$ Player | CombatDamage$ True | Execute$ TrigPeek | TriggerZones$ Battlefield | TriggerDescription$ Whenever CARDNAME deals combat damage to a player, look at the top card of your library. You may play a land from the top of your library or cast a spell with mana value less than or equal to the damage dealt from the top of your library without paying its mana cost. If you don't, put that card into your hand.
+SVar:TrigPeek:DB$ PeekAndReveal | PeekAmount$ 1 | NoReveal$ True | RememberPeeked$ True | SubAbility$ DBPlayLand
+SVar:DBPlayLand:DB$ Play | Optional$ True | ValidZone$ Library | Valid$ Land.IsRemembered | ForgetPlayed$ True | SubAbility$ DBCastSpell
```
Comment (tool4ever) [inline forge-gui/res/cardsfolder/upcoming/ziatoras_envoy.txt:8]: Did it not work with a single PlayEffect?

Comment (Northmoc) [reply forge-gui/res/cardsfolder/upcoming/ziatoras_envoy.txt:8]: Actually didn't try that... thought I'd need ValidSA for spells... I'll give it a try with one line!

Comment (Hanmac) [reply forge-gui/res/cardsfolder/upcoming/ziatoras_envoy.txt:8]: i really need to make LandAbility work for ValidSA

Comment (tool4ever) [reply forge-gui/res/cardsfolder/upcoming/ziatoras_envoy.txt:8]: Hmn, I thought SpellAbility.Land might work

**Review by Hanmac — APPROVED** [review]
👍

**Pull request merged**